### PR TITLE
Attempt to Fix Broken C++ Build After `cudftestutil` is Made a Shared Lib

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR aims to fix broken C++ build.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -285,7 +288,7 @@ if(Doxygen_FOUND)
 add_custom_command(
   OUTPUT CUSPATIAL_DOXYGEN
   WORKING_DIRECTORY ${CUSPATIAL_SOURCE_DIR}/doxygen
-  COMMAND ${DOXYGEN_EXECUTABLE} Doxyfile
+  COMMAND doxygen Doxyfile
```
[the remark this answers is not in the document]
Comment (isVoid) [reply cpp/CMakeLists.txt:291]: Is this how cudf finds doxygen too? I tried to look into https://github.com/rapidsai/cudf/blob/branch-23.04/cpp/CMakeLists.txt but cannot find any.

File: cpp/cmake/thirdparty/get_gtest.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# =============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
```
Comment (isVoid) [inline cpp/cmake/thirdparty/get_gtest.cmake:2]:
```suggestion
# Copyright (c) 2023, NVIDIA CORPORATION.
```

File: cpp/cmake/thirdparty/get_gtest.cmake [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# This function finds gtest and sets any additional necessary environment variables.
+function(find_and_configure_gtest)
+  include(${rapids-cmake-dir}/cpm/gtest.cmake)
+
+  # Find or install GoogleTest
+  rapids_cpm_gtest(BUILD_EXPORT_SET cudf-testing-exports INSTALL_EXPORT_SET cudf-testing-exports)
```
Comment (bdice) [inline cpp/cmake/thirdparty/get_gtest.cmake:20]: Should these be cuspatial?
```suggestion
  rapids_cpm_gtest(BUILD_EXPORT_SET cuspatial-testing-exports INSTALL_EXPORT_SET cuspatial-testing-exports)
```

File: cpp/cmake/thirdparty/get_gtest.cmake [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      BUILD GTest
+      VERSION ${GTest_VERSION}
+      EXPORT_SET GTestTargets
+      GLOBAL_TARGETS gtest gmock gtest_main gmock_main
+      NAMESPACE GTest::
+    )
+
+    include("${rapids-cmake-dir}/export/find_package_root.cmake")
+    rapids_export_find_package_root(
+      BUILD GTest [=[${CMAKE_CURRENT_LIST_DIR}]=] cudf-testing-exports
```
Comment (bdice) [inline cpp/cmake/thirdparty/get_gtest.cmake:33]:
```suggestion
      BUILD GTest [=[${CMAKE_CURRENT_LIST_DIR}]=] cuspatial-testing-exports
```

**Review by bdice — APPROVED** [review]
Looks fine if you fix the copied cudf references.

**Review by harrism — APPROVED** [review]

**Review by msadang — APPROVED** [review]
LGTM

**Pull request merged**